src/core/service.c

```diff
@@ -1250,23 +1271,98 @@ static int service_coldplug(Unit *u) {
         return 0;
 }
 
+static int connect_unix_probe(Service *s, const OpenFile *of, int ofd) {
```

Comment:
"probe" sounds like a misnomer: after all this actually does more than probing: it connects.

Maybe call this connext_unix_harder()?